the-collab-lab/tcl-30-whats-near-me
# [TCL30-15] Allow users to search for a location

## Description


This PR allows the user the ability to search locations by name or address to see wikipedia entries near locations they're interested in other than their own.  This functionality comes from the [Google Maps Places Library](https://developers.google.com/maps/documentation/javascript/examples/place-search), specifically using the `findPlaceFromQuery` method which returns a `name` and `geometry` allowing us to update the `centerMap` coordinates parameter for fetching nearby wikipedia entries.

## Related Issue


closes [TCL30-15](https://the-collab-lab.atlassian.net/browse/TCL30-15)

## Acceptance Criteria



- A search field should be present on both the map and table views
- Users should be able to make free text searches
- The user will be able to enter a place name or address, then click/tap a button to initiate the search
- The map will then center on the first result in the results object returned from the Places API
- If there are no results (e.g., if the user enters `gibberish` in the field), alert the user that no matching results were found, using the built in JavaScript alert method

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before


<img width="360" alt="Screen Shot 2021-08-15 at 2 35 01 PM" src="https://user-images.githubusercontent.com/2355543/129493584-cb104280-3ee1-46b0-8d5f-9cdda6543025.png">


### After


<img width="360" alt="Screen Shot 2021-08-15 at 2 37 16 PM" src="https://user-images.githubusercontent.com/2355543/129493586-d3b05166-3322-4783-8d2e-f7071a0a0d95.png">


## Testing Steps / QA Criteria


Note.- This time you can review our PR (at least app functionality) only with the PR link generated. However, you can see our code in our branch, please.

- From your terminal, pull down this branch `mm-ac-search-for-a-location`
- Run `npm start` to launch the app
- Verify that a search bar is visible on the `Map` (home page) and the `List` page
- Enter a location name and/or address into the search field and verify the wiki results for that area populate the `Map` page and the `List` page
- Enter a nonsense word or string to verify that an `alert` is triggered that states  'No matching results'
